packages/adminrouter/extra/src/includes/server/master.conf

```diff
@@ -264,13 +264,26 @@ location ~ ^/(slave|agent)/(?<agentid>[0-9a-zA-Z-]+)(?<url>.+)$ {
 
     more_clear_input_headers Accept-Encoding;
     include includes/proxy-headers.conf;
+
     # Non-streaming endpoints don't require HTTP/1.1 but will work as
     # expected with it enabled. Streaming endpoints require keepalive
     # functionality added in HTTP/1.1. As such, we enable HTTP/1.1 here
     # while maintaining backwards compatibility.
     include includes/http-11.conf;
     # Disable buffering to support streaming endpoints
     include includes/disable-request-response-buffering.conf;
+
+    # For the reasons stated in
+    # https://jira.mesosphere.com/browse/DCOS-42098
+    # we are setting keepalive timeout to 0, basically
+    # disabling it.
+
+    # ToDo (Arsen) Remove also the 'include includes/http-11.conf;'
+    # directive and deactivate the related harness tests.
+
+    # Disabling keepalive adds 'Connection: close' header to the
+    # response - https://forum.nginx.org/read.php?2,251560,251570
+    keepalive_timeout 0;
```

Comment:
Add issue to separate endpoints so streaming closes connection and non-streaming uses keepalive.

Reply:
Done: https://jira.mesosphere.com/browse/DCOS-42255

Reply:
I would actually love to see a little clearer commentary here.

```
    # For the reasons stated in
    # https://jira.mesosphere.com/browse/DCOS-42098
    # we are setting keepalive timeout to 0, basically
    # disabling it
```

Somehow this does not communicate the logic as clearly as we could communicate it :)

For the reasons stated in DCOS-42098 we want to make Admin Router emit a `Connection: close` header value. We use the `keepalive_timeout 0;` technique to do that.